src/approaches/nsrt_rl_approach.py

```diff
@@ -66,16 +84,121 @@ def learn_from_interaction_results(
         self._online_learning_cycle += 1
         # We get one result per training task.
         for i, result in enumerate(results):
-            states = result.states
-            actions = result.actions
-            traj = LowLevelTrajectory(states,
-                                      actions,
+            traj = LowLevelTrajectory(result.states,
+                                      result.actions,
                                       _is_demo=False,
                                       _train_task_idx=i)
```

Comment:
Well, we determine the order of the results because we define the method `get_interaction_requests()` for this class, which returns a list of requests. Then in `main.py`, the requests are handled sequentially. 

@ (1), we don't technically need the train task indices. We just need to associate the online trajectory with the plan (list of _Options) that was used to generate it. 
@ (2), InteractionResult already includes the train task index. 

I'll update the code to address these points.